File: day13/Ruby/factorial.rb
=begin
@author: aaditkamat
@date: 07/01/2019
=end

def factorial_recursive(num, result)
  if num < 0
    return "Factorial of #{num} is undefined"
  end
  if num === 0
    return result
  end
  factorial_recursive(num - 1, result * num)
end

def factorial_iterative(num)
  if num < 0
    return "Factorial of #{num} is undefined"
  end
  result = 1
  i = 1
  until i > num do
    result *= i
    i += 1
  end
  result
end

def main
  print "Enter an integer: "
  num = gets.chomp!.to_i
  puts "Factorial of #{num} is: #{factorial_recursive(num, 1)}"
  puts "Factorial of #{num} is: #{factorial_iterative(num)}"
end

main

File: day13/C#/factorial.cs
/**
 * @author: sagunjaiswal
 * @date: 20/09/2020
 */
using System; 
  
class day13 { 
    
    static int factorial(int n) 
    { 
        if (n == 0) 
            return 1; 
  
        return n * factorial(n - 1); 
    } 
  
    
    public static void Main() 
    { 
        String val;
        Console.WriteLine("Input : ");
        val = Console.ReadLine();
        
        int num = Convert.ToInt32(val);
        Console.WriteLine("Output:" + factorial(num)); 
    } 
} 